stdlib/public/runtime/Casting.cpp

```diff
@@ -169,7 +169,9 @@ void
 swift::swift_dynamicCastFailure(const void *sourceType, const char *sourceName, 
                                 const void *targetType, const char *targetName, 
                                 const char *message) {
+#if !defined(_WIN32)
   asm("");
```

Comment:
@rjmccall or @gparker42, do either of you recall why this is here? If it were intended as an optimization barrier, wouldn't it have to be `volatile`? As is, this would be a no-op even on Clang, AIUI.